contracts/service-registry/src/state.rs

```diff
@@ -300,6 +379,42 @@ pub fn deregister_chains_support(
     Ok(())
 }
 
+fn calculate_auth_verifier_count_change(
+    previous_state: Option<&AuthorizationState>,
+    auth_state: &AuthorizationState,
+) -> i16 {
+    match (previous_state, auth_state) {
+        (Some(NotAuthorized) | Some(Jailed), Authorized) | (None, Authorized) => 1,
+        (Some(Authorized), NotAuthorized) | (Some(Authorized), Jailed) => -1,
+        _ => 0,
```

Comment:
this is very brittle, if we add a new state the transition will not be covered and it will falsely fall into the 0 case. Please go with the suggestion I showed previously, explicitly stating `(Authorized, Authorized)`, `(_, Authorized)`, and `(Authorized, _)`. This will guarantee that future extensions wont cause an error